src/main/scala/fi/vm/sade/hakurekisteri/web/kkhakija/KkExcelUtilV5.scala

```diff
@@ -0,0 +1,236 @@
+  )
+
+  override def getRows(hakijat: Seq[Hakija]): Set[Row] = hakijat
+    .flatMap(hakija =>
+      hakija.hakemukset.map(hakemus => {
+        val rivi = Seq(
+          hakija.hetu,
+          hakija.syntymaaika.getOrElse(""),
+          hakija.oppijanumero,
+          hakija.sukunimi,
+          hakija.etunimet,
+          hakija.kutsumanimi,
+          hakija.lahiosoite,
+          hakija.postinumero,
+          hakija.postitoimipaikka,
+          hakija.maa,
+          hakija.kansalaisuudet.getOrElse(List.empty).mkString(", "),
+          hakija.matkapuhelin.getOrElse(""),
+          hakija.puhelin.getOrElse(""),
+          hakija.sahkoposti.getOrElse(""),
+          hakemus.lukuvuosimaksu.getOrElse(""),
+          hakija.kotikunta,
+          hakija.sukupuoli,
+          hakija.aidinkieli,
+          hakija.asiointikieli,
+          hakija.koulusivistyskielet.getOrElse(List.empty).mkString(", "),
+          toBooleanX(hakija.koulutusmarkkinointilupa),
+          toBooleanX(hakija.onYlioppilas),
+          hakija.yoSuoritusVuosi.getOrElse(""),
+          hakija.ensikertalainen.map(e => toBooleanX(e)).getOrElse(""),
+          hakemus.haku,
+          hakemus.hakuVuosi.toString,
+          hakemus.hakuKausi,
+          hakemus.hakemusnumero,
+          hakemus.hakemusJattoAikaleima.getOrElse(""),
+          hakemus.hakemusViimeinenMuokkausAikaleima.getOrElse(""),
+          hakemus.organisaatio,
+          hakemus.hakukohde,
+          hakemus.hakukohdeKkId.getOrElse(""),
+          hakemus.hakutoivePrioriteetti.map(_.toString).getOrElse(""),
+          toBooleanX(hakemus.avoinVayla),
+          hakemus.valinnanTila.map(_.toString).getOrElse(""),
+          hakemus.valinnanAikaleima.getOrElse(""),
+          hakemus.pisteet.map(_.toString()).getOrElse(""),
+          hakemus.hyvaksymisenEhto match {
+            case None => ""
+            case Some(h) =>
+              s"HyvaksymisenEhto(${toBooleanX(h.ehdollisestiHyvaksyttavissa)},${h.ehtoKoodi.getOrElse("")}," +
+                s"${h.ehtoFI.getOrElse("")},${h.ehtoSV.getOrElse("")},${h.ehtoEN.getOrElse("")})"
+          },
+          hakemus.valintatapajononTyyppi.getOrElse(""),
+          hakemus.valintatapajononNimi.getOrElse(""),
+          hakemus.vastaanottotieto.map(_.toString).getOrElse(""),
+          hakemus.ilmoittautumiset.mkString(","),
+          hakemus.pohjakoulutus.mkString(","),
+          toBooleanX(hakemus.julkaisulupa),
+          hakemus.hKelpoisuus,
+          hakemus.hKelpoisuusLahde.getOrElse(""),
+          hakemus.hKelpoisuusMaksuvelvollisuus.getOrElse(""),
+          hakemus.hakukohteenKoulutukset lift 0 match {
+            case Some(k) =>
+              s"Koulutus(${k.komoOid},${k.tkKoulutuskoodi},${k.kkKoulutusId.getOrElse("")}," +
+                s"${k.koulutuksenAlkamisvuosi.getOrElse("")},${k.koulutuksenAlkamiskausi.getOrElse("")}," +
+                s"${k.johtaaTutkintoon.getOrElse("")})"
```

Comment:
Tuli mieleen että näitä varten voisi ehkä lisätä tuolle KkHakukohteenkoulutus-case classille jonkinlaisen toString-henkisen apumetodin niin pääsisi toistosta

Reply:
Hyvä idea, lisäsin tuolle ja liitteille moiset. Käytössä versioimatta vain v5:ssä, aiemmissa nuo tiedot ovat vähän erilaiset.